packages/webpack-cli/lib/utils/get-package-manager.js

```diff
@@ -1,37 +1,33 @@
 const fs = require('fs');
 const path = require('path');
-const { sync } = require('execa');
 
 /**
  *
  * Returns the name of package manager to use,
- * preferring yarn over npm if available
+ * preference order - npm > yarn > pnpm
  *
  * @returns {String} - The package manager name
  */
 function getPackageManager() {
-    const hasLocalYarn = fs.existsSync(path.resolve(process.cwd(), 'yarn.lock'));
     const hasLocalNpm = fs.existsSync(path.resolve(process.cwd(), 'package-lock.json'));
-    const hasLocalPnpm = fs.existsSync(path.resolve(process.cwd(), 'pnpm-lock.yaml'));
+
+    if (hasLocalNpm) {
+        return 'npm';
+    }
+
+    const hasLocalYarn = fs.existsSync(path.resolve(process.cwd(), 'yarn.lock'));
 
     if (hasLocalYarn) {
         return 'yarn';
-    } else if (hasLocalNpm) {
-        return 'npm';
-    } else if (hasLocalPnpm) {
-        return 'pnpm';
     }
 
-    try {
-        // if the sync function below fails because yarn is not installed,
-        // an error will be thrown
-        if (sync('yarn', ['--version'])) {
-            return 'yarn';
-        }
```

Comment:
It seems mocking `execa.sync` isn't required anymore.

Reply:
yep removed.

Reply:
No, it was for projects where lock files disabled, it is rare case right now, but better support it

Reply:
just because `yarn` is installed does not mean users want to use `yarn`. I'm forced to have it installed for a project or two, but I have hundreds of other projects that use npm.

The default should be `npm` since that's the node's default.

Reply:
Ideally we have here more powerful logic, if we don't found lock files, we should start to check `npm -v`/`yarn -v`/`pnpm -v`, and if we not found something here too, we should throw an error - `No package managers found please install package manually`, it is possible when node was wtihout package manager, npm can be removed too

Reply:
I see so the order should be `global npm` > `global yarn` > `global pnpm` ?

Reply:
Yes, logic:
- search npm lock file
- search yarn lock file
- search pnpm lock file
- run `npm -v`
- run `yarn -v`
- run `pnpm -v`
- error - No package manager found